pkg/detectors/intra42/intra42.go

```diff
@@ -0,0 +1,117 @@
+package intra42
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time
+var _ detectors.Detector = (*Scanner)(nil)
+
+const verifyURL = "https://api.intra.42.fr/oauth/token"
+
+var (
+	defaultClient = common.SaneHttpClient()
+
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"intra", "intra42", "client", "secret"}) + `\b(s-s4t2(?:ud|af)-[abcdef0123456789]{64})\b`)
```

Comment:
suggestion: We could simplify the alphanumeric portion of the regex from `abcdef0123456789` -> `a-f0-9`. We could also make the same update below.

We can also likely remove the `detectors.PrefixRegex` here since the keywords are already a prefix that is part of the key.

Ex:
```
keyPat = regexp.MustCompile(`\b(s-s4t2(?:ud|af)-[a-f0-9]{64})\b`)
idPat  = regexp.MustCompile(`\b(u-s4t2(?:ud|af)-[a-f0-9]{64})\b`)
```

Reply:
Totally agree for both! I went with PrefixRegex because the example used it, without thinking of removing it. Will update.